.github/ISSUE_TEMPLATE/release-check-list.md

```diff
@@ -8,106 +8,110 @@ assignees: ''
-      * Change the revision
-      * Run `cargo update -p image-rs`
-
-- [ ] 2. Update Kata Containers to use the latest commit from image-rs, attestation-agent and td-shim
+## Overview
+
+The release process mainly follows from this dependency graph.
+
+```mermaid
+flowchart LR
+    Trustee --> Versions.yaml
+    Guest-Components --> Versions.yaml
+    Kata --> kustomization.yaml
+    Guest-Components .-> Client-tool
+    Guest-Components --> enclave-agent
+    enclave-cc --> kustomization.yaml
+    Guest-Components --> versions.yaml
+    Trustee --> versions.yaml
+    Kata --> versions.yaml
+
+    subgraph Kata
+        Versions.yaml
+    end
+    subgraph Guest-Components
+    end
+    subgraph Trustee
+        Client-tool
+    end
+    subgraph enclave-cc
+        enclave-agent
+    end
+    subgraph Operator
+        kustomization.yaml
+        reqs-deploy
+    end
+    subgraph cloud-api-adaptor
+        versions.yaml
+    end
+```
+
+Starting wth v0.9.0 the release process no longer involves centralized dependency management.
+In other words, when doing a CoCo release, we don't push the most recent versions of the subprojects
+into Kata and enclave-cc. Instead, dependencies should be updated during the normal process of development.
+Releases of most subprojects are now decoupled from releases of the CoCo project.
+
+## The Steps
+
+### Determine release builds
+
+The release starts by testing Kata and Enclave-CC with the most recent versions
+of all of their CoCo dependencies.
+
+- [ ] 0. :snowman: **Code Freeze**
+
+    At this point, all CoCo development should cease until the release is complete.
+    Post a note in the Slack channel about this.
+    Unrelated work may continue in the Kata repo as long as the CI stays green.
+
+- [ ] 1. :eyes: **Find enclave-cc release version**
+
+- [ ] 2. :eyes: **Find Kata release version**
+
+
+### Test Release with Operator
+
+- [ ] 3. :eyes: **Check operator pre-installation**
+    
+    The operator uses a pre-install container to setup the node.
+    Check that the container matches the depdencies used in Kata
+    and that the operator pulls the most recent version of the container.
 
-    * image-rs
-      * https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/Cargo.toml
-      * Change the revision
-      * Run `cargo update -p image-rs`
-    * attestation-agent and td-shim
-      * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
-        * Change the version
+    * Check that the version of the `nydus-snapshotter` used by Kata matches the one used by the operator
+        * Compare `nydus-snapshotter` version in Kata [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L291) [this]() with the [Makefile](https://github.com/confidential-containers/operator/blob/main/install/pre-install-payload/Makefile#L4) for the opertor pre-intall container.
```

Comment:
```suggestion
        * Compare `nydus-snapshotter` version in Kata [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L291) with the [Makefile](https://github.com/confidential-containers/operator/blob/main/install/pre-install-payload/Makefile#L4) for the opertor pre-intall container.
```

Reply:
Are we planning to do this nydus-snapshotter version bump for the 0.9.0 release? I think it's a good idea but also falls under the category of something we should try and do ahead of time, so that when kata's version increases we do it on the operator then rather than waiting for release time?

Reply:
I think future releases are going to change this pre-install flow a bit. Probably the operator and kata should always be in sync with the version but I'm not sure the best process for that. I figure we should still check that the versions are the same during the release even if we try to update them earlier as well.

Reply:
btw I fixed the typo that @mythi first pointed out